mllib/src/main/scala/org/apache/spark/ml/feature/Word2Vec.scala

```diff
@@ -105,6 +106,56 @@ private[feature] trait Word2VecBase extends Params
   /** @group getParam */
   def getMaxSentenceLength: Int = $(maxSentenceLength)
 
+  /**
+   * Number of negative samples to use with CBOW based estimation.
+   * This parameter is ignored for SkipGram-Hierachical Softmax based estimation.
+   * Default: 15
+   * @group param
+   */
+  final val numNegativeSamples = new IntParam(this, "numNegativeSamples", "Number of negative" +
+    " samples to use with CBOW estimation", ParamValidators.gt(0))
+  setDefault(numNegativeSamples -> 15)
+
+  /** @group getParam */
+  def getNumNegativeSamples: Int = $(numNegativeSamples)
+
+  /**
+   * Unigram table size. The unigram table is used to generate negative samples.
+   * This parameter is ignored for SkipGram based estimation.
+   * Default: 100 million
+   * @group param
+   */
+  final val unigramTableSize = new IntParam(this, "unigramTableSize", "Max table size to " +
+    "use for generating negative samples", ParamValidators.gt(1))
+  setDefault(unigramTableSize -> 100*1000*1000)
```

Comment:
maybe move the setDefault after setter in Word2Vec, which seems to be a recommended practice right now. (Just as the solver)